test/integration/MCR/updateMCR.js

```diff
@@ -173,86 +185,78 @@ describe('updateMCR', function () {
       .div(parseEther('1'))
       .div(ratioScale);
 
-    // Cover inputs
-    const productId = 0;
-    const coverAsset = 0; // ETH
-    const period = daysToSeconds(30);
-    const gracePeriod = 3600 * 24 * 30;
-    const amount = coverAmount;
-
-    // Stake to open up capacity
-    await stake({ stakingPool: stakingPool1, staker: staker1, gracePeriod, period, productId });
-
-    const expectedPremium = parseEther('1');
-    await cover.connect(coverHolder).buyCover(
-      {
-        coverId: 0,
-        owner: coverHolder.address,
-        productId,
-        coverAsset: 0,
-        amount,
-        period,
-        maxPremiumInAsset: expectedPremium,
-        paymentAsset: coverAsset,
-        commissionRatio: parseEther('0'),
-        commissionDestination: ethers.constants.AddressZero,
-        ipfsData: '',
-      },
-      [{ poolId: 1, coverAmountInAsset: amount }],
-      { value: expectedPremium },
-    );
+    // buy cover
+    const newCoverBuyParams = {
+      ...newEthCoverTemplate,
+      amount: coverAmount,
+      cover,
+      coverBuyer: coverHolder,
+      targetPrice: this.DEFAULT_PRODUCTS[0].targetPrice,
+      expectedPremium: parseEther('100'),
+    };
+    await buyCover({
+      ...newCoverBuyParams,
+    });
 
     const lastUpdateTimeBefore = await mcr.lastUpdateTime();
-    await increaseTime(await mcr.minUpdateTime());
-
     const mcrFloorBefore = await mcr.mcrFloor();
-
+    await increaseTime(await mcr.minUpdateTime());
     await mcr.updateMCR();
+
     const { timestamp: currentTime } = await ethers.provider.getBlock('latest');
     const lastUpdateTimeAfter = await mcr.lastUpdateTime();
     const mcrFloorAfter = await mcr.mcrFloor();
     const desireMCRAfter = await mcr.desiredMCR();
     const expectedDesiredMCR = coverAmount.div(gearingFactor).mul(ratioScale);
 
-    assert(lastUpdateTimeBefore < lastUpdateTimeAfter);
+    expect(lastUpdateTimeBefore).to.be.lt(lastUpdateTimeAfter);
     expect(lastUpdateTimeAfter).to.be.equal(currentTime);
-    assert(
-      mcrFloorAfter.gt(mcrFloorBefore),
+    expect(mcrFloorAfter).to.be.gt(
+      mcrFloorBefore,
       `MCR floor post update ${mcrFloorAfter.toString()} is not greater than before ${mcrFloorBefore.toString()}`,
     );
+    // TODO: this assertion is off
```

Comment:
AssertionError: expected 40016000000000000000000 to equal 50000. The numerical values of the given "ethers.BigNumber" and "ethers.BigNumber" inputs were compared, and they differed.
      + expected - actual

      -40016000000000000000000
      +50000

Reply:
So I fixed the magnitude issue. This one is also off by a small amount though :thinking:

Reply:
Here is the last part of the stack trace for the last `updateMCR()` call.
```
 UPDATE MCR 


desiredMCR 50000000000000000000000
hours since update 1
CALL MCR(0x95401dc811bb5740090279ba06cfa8fcf6113778).updateMCR()
   STATICCALL Pool(0x4826533b4897376654bb4d4ad88b7fafd0c98528).getPoolValueInEth() => (110198453836301282947051)
      STATICCALL MOCK(0xe7f1725e7734ce288f8367e1bb143e90bb3f0512).balanceOf(account: [Pool]) => (2000000.0)
      STATICCALL PriceFeedOracle(0x70e0ba845a1a0f2da3359c97e0285013525ffc49).getEthForAsset(asset: [MOCK], amount: 2000000000000000000000000) => (10000000000000000000000)
   STATICCALL Pool(0x4826533b4897376654bb4d4ad88b7fafd0c98528).calculateMCRRatio(totalAssetValue: 110198453836301282947051, mcrEth: 50000000000000000000000) => (22039)

   STATICCALL OwnedUpgradeabilityProxy(0xfd471836031dc5108809d173a067e8486b9047a3).totalActiveCoverInAsset(coverAsset: 0) => (241440000833227585579221)
   EVENT ICover(0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266).MCRUpdated(mcr: 50000000000000000000000, desiredMCR: 50300000173589080329004, mcrFloor: 40016000000000000000000, mcrETHWithGear: 50300000173589080329004, totalSumAssured: 241440000833227585579221)
CALL MCR(0x95401dc811bb5740090279ba06cfa8fcf6113778).lastUpdateTime() => (1677285517)
CALL MCR(0x95401dc811bb5740090279ba06cfa8fcf6113778).mcrFloor() => (40016000000000000000000)
CALL MCR(0x95401dc811bb5740090279ba06cfa8fcf6113778).desiredMCR() => (50300000173589080329004)
          1) increases desiredMCR if totalSumAssured is high enough

      AssertionError: expected 50300000173589080329004 to equal 50300000000000000000000. The numerical values of the given "ethers.BigNumber" and "ethers.BigNumber" inputs were compared, and they differed.
      + expected - actual

      -50300000173589080329004
      +50300000000000000000000
      
      at Context.<anonymous> (test/integration/MCR/updateMCR.js:227:34)

```

Reply:
this looks  more tolerable we still need to see exactly why bc we should replicate this in js